framework/src/Volo.Abp.Core/Volo/Abp/Internal/Telemetry/Activity/Providers/TelemetryActivityEventEnricher.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.DynamicProxy;
+using Volo.Abp.Internal.Telemetry.Activity.Contracts;
+
+namespace Volo.Abp.Internal.Telemetry.Activity.Providers;
+
+public abstract class TelemetryActivityEventEnricher : ITelemetryActivityEventEnricher, IScopedDependency
+{
+    public virtual int ExecutionOrder { get; set; } = 0;
+    protected bool CancelChildren { get; set; }
+    protected virtual Type? ReplaceParentType { get; set; }
+
+    private readonly IServiceProvider _serviceProvider;
+
+    protected TelemetryActivityEventEnricher(IServiceProvider serviceProvider)
+    {
+        _serviceProvider = serviceProvider;
+    }
+
+    public async Task EnrichAsync(ActivityContext context)
+    {
+        if (!await CanExecuteAsync(context))
+        {
+            return;
+        }
+
+        await ExecuteAsync(context);
+        await ExecuteChildrenAsync(context);
+    }
+
+    protected virtual Task<bool> CanExecuteAsync(ActivityContext context)
+    {
+        return Task.FromResult(true);
+    }
+
+    protected abstract Task ExecuteAsync(ActivityContext context);
+
+    private async Task ExecuteChildrenAsync(ActivityContext context)
```

Comment:
Making it `protected virtual` might be better, incase we need to override it later on enrichers (we may have additional logics for children):

```suggestion
    protected virtual async Task ExecuteChildrenAsync(ActivityContext context)
```